solutions/cses-1160.mdx

```diff
@@ -33,106 +35,112 @@ For the third and fourth case, one possible optimization is to use binary liftin
 #include <bits/stdc++.h>
 using namespace std;
 
-int N, Q, cycle_len[200005];
+const int MAXN = 2e5 + 1;
+const int LOGN = 20;
+
+int N, Q, cycle_len[MAXN];
 
-struct node {
-	int par[20], cycle_id = 0, dis = 0, vis = 0;
+struct Node {
+	int par[LOGN], cycle_id = 0, dis = 0, vis = 0;
 	bool cycle = 0;
-} nde[200005];
+} node[MAXN];
 
 pair<int, int> dfs(int u, int dfs_id) {
-	if (nde[u].vis) {
-		if (nde[u].vis == dfs_id) {  // cycle case
+	if (node[u].vis) {
+		if (node[u].vis == dfs_id) {  // cycle decomposition
```

Comment:
you talked about this crap in the editorial right

Reply:
I don't understand how I did this

Reply:
I sort of glossed over it

Reply:
bruh